### PR TITLE
Keep order of read from segment consist with the write order

### LINKED ISSUE #3989: UNIQUE_KEYS table data may not stable when user load data with multiple same key lines
When importing data into UNIQUE_KEYS table , if there are rows with the same key in the data, when the amount of data is large, the rows with the same key may be distributed in different segments. The priority_queue used when reading the rowset is an unstable sort, which may occur the lastest imported data row is read first, resulting is inconsistent between the final state of the data and the imported state。
The comparator of merge only compare the row not include segment id.
image

### PR DESCRIPTION
**Pull request by yangzhg**
Fixes #3989 
Add segemt id to the comparator

### CODE REVIEW

File: be/src/olap/generic_iterators.cpp [hunk 1]
```diff
@@ -227,8 +229,13 @@ class MergeIterator : public RowwiseIterator {
         bool operator()(const MergeIteratorContext* lhs, const MergeIteratorContext* rhs) const {
             auto lhs_row = lhs->current_row();
             auto rhs_row = rhs->current_row();
-
-            return compare_row(lhs_row, rhs_row) > 0;
+            int cmp_res = compare_row(lhs_row, rhs_row);
+            if (cmp_res !=  0) {
+                return cmp_res > 0;
+            }
+            // if row cursors equal, compare segment id.
```
Comment (wutiangan) [inline be/src/olap/generic_iterators.cpp:236]: add some comments why  sort by row asc, and sort by segment_id desc.

**Comment by chaoyli** [comment]
Do you can reproduce the problem using UNIQUE KEY?
Because I have a little double why this bug exists for so long time.

**Comment by yangzhg** [comment]
@chaoyli this problem is really exists for a long time. the issue #3930 describe the load cannot keep order and this pr fix read cannot keep order

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**